### PR TITLE
feat(vuln): remove duplicates in Fixed Version

### PR DESCRIPTION
**Pull request by ilpianista**
## Description

Sometime, the Fixed Version column may contains the same versions repeatedly:

```
Python (python-pkg)
===================
Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 1)

┌───────────────────────┬────────────────┬──────────┬────────┬───────────────────┬──────────────────────────────────────────────┬──────────────────────────────────────────────────────────────┐
│        Library        │ Vulnerability  │ Severity │ Status │ Installed Version │                Fixed Version                 │                            Title                             │
├───────────────────────┼────────────────┼──────────┼────────┼───────────────────┼──────────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ Django (METADATA)     │ CVE-2023-31047 │ CRITICAL │ fixed  │ 4.0.10            │ 3.2.19, 4.1.9, 4.2.1                         │ Potential bypass of validation when uploading multiple files │
│                       │                │          │        │                   │                                              │ using one form field...                                      │
│                       │                │          │        │                   │                                              │ https://avd.aquasec.com/nvd/cve-2023-31047                   │
│                       ├────────────────┼──────────┤        │                   ├──────────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                       │ CVE-2023-36053 │ HIGH     │        │                   │ 3.2.20, 4.1.10, 4.2.3, 4.2.3, 4.1.10, 3.2.20 │ Potential regular expression denial of service vulnerability │
│                       │                │          │        │                   │                                              │ in EmailValidator/URLValidator                               │
│                       │                │          │        │                   │                                              │ https://avd.aquasec.com/nvd/cve-2023-36053                   │
└───────────────────────┴────────────────┴──────────┴────────┴───────────────────┴──────────────────────────────────────────────┴──────────────────────────────────────────────────────────────┘
```

AFTER:

```
┌───────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────────────┬──────────────────────────────────────────────────────────────┐
│        Library        │ Vulnerability  │ Severity │ Status │ Installed Version │     Fixed Version     │                            Title                             │
│                       ├────────────────┼──────────┼────────┼───────────────────┼───────────────────────┼──────────────────────────────────────────────────────────────┤
│ Django (METADATA)     │ CVE-2023-31047 │ CRITICAL │ fixed  │ 4.0.10            │ 3.2.19, 4.1.9, 4.2.1  │ Potential bypass of validation when uploading multiple files │
│                       │                │          │        │                   │                       │ using one form field...                                      │
│                       │                │          │        │                   │                       │ https://avd.aquasec.com/nvd/cve-2023-31047                   │
│                       ├────────────────┼──────────┤        │                   ├───────────────────────┼──────────────────────────────────────────────────────────────┤
│                       │ CVE-2023-36053 │ HIGH     │        │                   │ 3.2.20, 4.1.10, 4.2.3 │ Potential regular expression denial of service vulnerability │
│                       │                │          │        │                   │                       │ in EmailValidator/URLValidator                               │
│                       │                │          │        │                   │                       │ https://avd.aquasec.com/nvd/cve-2023-36053                   │
└───────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────────────┴──────────────────────────────────────────────────────────────┘
```

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by ilpianista** [comment]
@knqyf263 any chance you have time to review this? :pray:

**Comment by ilpianista** [comment]
> Could you fix tests first?

Done, sorry, I didn't realize that.

**Comment by DmitriyLewen** [comment]
Hm... this is strange case.
GitHub contains 2 same packages for [CVE-2023-36053](https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2023/07/GHSA-jh3w-4vvf-mjgr/GHSA-jh3w-4vvf-mjgr.json) (Django and django (lower and upper case for 1st character))

File: pkg/detector/library/driver.go [hunk 1]
```diff
@@ -135,19 +135,31 @@ func (d *Driver) DetectVulnerabilities(pkgID, pkgName, pkgVer string) ([]types.D
...
-
 	var fixedVersions []string
-	for _, version := range advisory.VulnerableVersions {
-		for _, s := range strings.Split(version, ",") {
-			s = strings.TrimSpace(s)
-			if !strings.HasPrefix(s, "<=") && strings.HasPrefix(s, "<") {
-				s = strings.TrimPrefix(s, "<")
-				fixedVersions = append(fixedVersions, strings.TrimSpace(s))
+
+	uniqueVersions := make(map[string]int)
```
Comment (DmitriyLewen) [inline pkg/detector/library/driver.go:140]: I think we can just use [lo.Uniq(fixedVersions)](https://pkg.go.dev/github.com/samber/lo#readme-uniq)

**Review by DmitriyLewen — COMMENTED** [review]
Hello @ilpianista 
Thanks for your work!

I left 1 comment.

Can you also add a small test case to `driver_test.go` to make sure your changes work.

File: pkg/detector/library/driver.go [hunk 2]
```diff
@@ -135,19 +136,22 @@ func (d *Driver) DetectVulnerabilities(pkgID, pkgName, pkgVer string) ([]types.D
...
+				if !strings.HasPrefix(s, "<=") && strings.HasPrefix(s, "<") {
+					s = strings.TrimPrefix(s, "<")
+					fixedVersions = append(fixedVersions, strings.TrimSpace(s))
+				}
 			}
 		}
 	}
-	return strings.Join(fixedVersions, ", ")
+
+	return strings.Join(lo.Uniq(fixedVersions), ", ")
```
Comment (DmitriyLewen) [inline pkg/detector/library/driver.go:156]: If I remember correctly, `lo.Uniq` doesn't preserve order.
Can you test this and add another version range to the new test case to make sure we don't break the order of the versions (I mean keeping the versions in ascending order)?

Comment (ilpianista) [reply pkg/detector/library/driver.go:156]: It does:
> The order of result values is determined by the order they occur in the array.

**Pull request merged**